### DISCUSSION TITLE
GitHub pages loading random CSS file. Different to local build

### DISCUSSION THREAD

**Discussion by courtneycb** [community/community#22252] [Other Feature Feedback, Questions, & Ideas]
<p>Hi there!</p>
<p>I’m running a Jekyll site and everything is looking sweet locally but when the site is deployed to GH Pages some random CSS file is loaded instead of the one I have supplied. I have disabled any jekyll themes. Here is my repo: <a href="https://github.com/uccser/ci-dashboard" rel="noopener nofollow ugc">https://github.com/uccser/ci-dashboard</a></p>
<p>I have this link to my style sheet in my HTML:<br>
<code>&lt;link rel="stylesheet" href="/assets/css/style.css"&gt;</code></p>
<p>and I have put this in my _config.yaml file:</p>
<pre><code class="lang-auto">theme: null

sass:
  sass_dir: assets/css/
</code></pre>
<p>When the site is published to GH pages the css file it loads is this:<br>
<img src="https://user-images.githubusercontent.com/16066822/181069734-373eba42-5bb5-4726-989a-2650964f289a.png" alt="Screen Shot 2020-11-19 at 3.24.24 PM" data-base62-sha1="14TUJZCyznJ6kzkjp04vIZbeM8z" width="619" height="291"></p>
<p>But locally it loads like this:<br>
<img src="https://user-images.githubusercontent.com/16066822/181069735-cc9b5251-c22a-4505-9e32-737fd7005efa.png" alt="Screen Shot 2020-11-19 at 3.25.51 PM" data-base62-sha1="jmVfHbPVQpbytFZ5XOlad6AIHCq" width="618" height="288"></p>
<p>None of the css I have supplied in my own file loads and I have no idea where the rules in the css file on GH Pages have come from.</p>
<p>Local:<br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/d/c/dcbcad33946d078642c5c60534d7040362c3dc3a.png" data-download-href="/uploads/short-url/vuJhqRyimoyBfRHNtUZXifQyD1g.png?dl=1" title="Screen Shot 2020-11-19 at 3.29.09 PM" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/16066822/181069736-6ac12e79-7f0e-448f-9f5e-66dc31c6327b.png" alt="Screen Shot 2020-11-19 at 3.29.09 PM" data-base62-sha1="vuJhqRyimoyBfRHNtUZXifQyD1g" width="690" height="183" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/d/c/dcbcad33946d078642c5c60534d7040362c3dc3a_2_690x183.png, https://aws1.discourse-cdn.com/github/optimized/3X/d/c/dcbcad33946d078642c5c60534d7040362c3dc3a_2_1035x274.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/d/c/dcbcad33946d078642c5c60534d7040362c3dc3a.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/d/c/dcbcad33946d078642c5c60534d7040362c3dc3a_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screen Shot 2020-11-19 at 3.29.09 PM</span><span class="informations">1250×332 22.4 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>GH Pages:<br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/9/c/9c67b1ff87c86a8d3ca8f42faf3016b4c15a1895.png" data-download-href="/uploads/short-url/mjCEoAiOgLk2I5MjH6015Quz565.png?dl=1" title="Screen Shot 2020-11-19 at 3.29.30 PM" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/16066822/181069737-4e5f3adc-eb5e-4f03-84ef-fe84fbc0202c.png" alt="Screen Shot 2020-11-19 at 3.29.30 PM" data-base62-sha1="mjCEoAiOgLk2I5MjH6015Quz565" width="690" height="212" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/9/c/9c67b1ff87c86a8d3ca8f42faf3016b4c15a1895_2_690x212.png, https://aws1.discourse-cdn.com/github/optimized/3X/9/c/9c67b1ff87c86a8d3ca8f42faf3016b4c15a1895_2_1035x318.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/9/c/9c67b1ff87c86a8d3ca8f42faf3016b4c15a1895.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/9/c/9c67b1ff87c86a8d3ca8f42faf3016b4c15a1895_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screen Shot 2020-11-19 at 3.29.30 PM</span><span class="informations">1219×376 23.1 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by otter-computer** [comment, marked as answer]
<p>It looks like you’re running into a similar pathing issue as listed in this answer here:</p>
<aside class="quote quote-modified" data-post="5" data-topic="136703">
  <div class="title">
    <div class="quote-controls"></div>
    
    <a href="https://github.community/t/code-showing-correctly-in-preview-but-not-actual-page/136703/5">Code showing correctly in preview, but not actual page?</a> <a class="badge-wrapper  bullet" href="/c/github-help/github-pages/31"><span class="badge-category-parent-bg" style="background-color: #6a737d;"></span><span class="badge-category-bg" style="background-color: #c8e1ff;"></span><span style="" data-drop-close="true" class="badge-category clear-badge" title="Discussion and questions about GitHub Pages—websites for you and your projects hosted right on your GitHub repository.">GitHub Pages</span></a>
  </div>
  <blockquote>
    Thanks for posting those links! The reason this is happening is because of a relative path issue with how the browser loads this image file on <a href="http://GitHub.com">GitHub.com</a> compared to on your Pages site. Looking at the code for this image: 
&lt;img src="/images/demo/meeting.jpg"&gt;

The source for this image starts with a / character. This tells the browser to search for the image file starting at the root of the domain. When you’re browsing <a href="http://GitHub.com">GitHub.com</a> this means it will look in the root of the repository then follow …
  </blockquote>
</aside>

<p>You’ve referenced your CSS import like this:</p>
<pre><code class="lang-auto">&lt;link rel="stylesheet" href="assets/css/style.css"&gt;
</code></pre>
<p>The URL you want the browser to be loading is the following:</p>
<pre><code class="lang-nohighlight">https://uccser.github.io/ci-dashboard/assets/css/style.css
</code></pre>
<p>However, it’s <em>actually</em> loading this URL—the CSS file for another site hosted under your account:</p>
<pre><code class="lang-nohighlight">https://uccser.github.io/assets/css/style.css
</code></pre>
<p>To fix this you’ll need to make sure that Jekyll adds your repository name to your link when building your site. You can do this by setting the <code>baseurl</code> value inside your config file</p>
<pre><code class="lang-auto">baseurl: /ci-dashboard
</code></pre>
<p>You’ll then need to update your CSS link to tell Jekyll to prepend this baseurl value to your links:</p>
<pre><code class="lang-auto">&lt;link rel="stylesheet" href="{{ "assets/css/style.css" | relative_url }}"&gt;
</code></pre>

**Comment by courtneycb** [comment]
<p><a class="mention" href="/thomasshaped">@thomasshaped</a> thank you SO MUCH!!! You’re amazing.</p>
